DOC: legal process to perform a public release of a PyAnsys library (#236)
pyansys/dev-guide: doc/source/getting-started/administration.rst

Project approval and public release
===================================
Most of the projects in PyAnsys expose the functionality of Ansys
products. Due to intellectual property reasons, the public release of a PyAnsys
library must go through a project approval process.

Approval process
----------------
The approval process is divided into three parts:

.. grid:: 3
    
    .. grid-item-card:: :octicon:`file-badge` Managerial
       :link: managerial
       :link-type: ref

>         Verifies that direct managers, general managers, or chieff technology

Reviewer: ```suggestion
        Verifies that direct managers, general managers, or chief technology
```

>         Verifies that direct managers, general managers, or chief technology
>         officer are aware of the project and approve it.

Reviewer 2: ```suggestion
        Verifies that direct managers, general managers, and chief technology
        officer are aware of the project and approve it.
```

    :color: black
    :expand:

    **Open Source Code Release Request Form**

The following checks are required when performing the legal review of the project:

.. card:: |uncheck| The project contains the right licensing.

    * The project contains the right license.
    * The contribution does not contain any strong encryption.
    * Ansys official logos and branding images are used in the project.
    * The Ansys copyright appears in the right location as required by the Legal department.
    * The copyright has the right formatting, which is ``Copyright (C) YYYY ANSYS, Inc.``.
    * The contribution does not embody any Ansys intellectual property that is not approved for open sourcing.
    * The contribution does not embody any invention for which Ansys has sought or received patent protection.
    * Any third-party open sources included in the contribution have been reviewed for security vulnerabilities and have had their license files included in the repository.

Open source dependencies that are not distributed as part of the project do not
need their licenses included in the Ansys repository. Examples include

> dependent Node package manager (npm) modules or Python packages from PyPI.

Reviewer: ```suggestion
dependent Node package manager (``npm``) modules or Python packages from PyPI.
```

Project approval and public release
===================================

> Most of the projects in PyAnsys expose the functionality of ANSYS, Inc.

Author: I thought about using `ANSYS, Inc.` here because the content was related to legal stuff. Should I go with the usual `Ansys` name instead, @PipKat?

> Most of the projects in PyAnsys expose the functionality of ANSYS, Inc.

Author: As pointed out by Kathy, let us use "Ansys" since this is not a copyright-related document.

products. Due to intellectual property reasons, the public release of a PyAnsys

> library must undergo through a project approval process.

Reviewer 3: Either "...must undergo a project approval process" or "...must go through a project approval process".

Approval process
----------------

> The approval process is divided into three different parts:

Reviewer 3: ```suggestion
The approval process is divided into three parts:
```

.. grid:: 3
    
    .. grid-item-card:: :octicon:`file-badge` Managerial
       :link: managerial
       :link-type: ref

>         Verifies that direct managers, GM/VP, CT or CTO are aware about the project and approve it.

Author: I decided to use the abbreviations here since this abstract text appears inside a card. Thus, the available space is limited. Once users click, they get redirected to the corresponding section, where all previous abbreviations are properly explained.

Hence, should we leave this as it is?

    .. grid-item-card:: :octicon:`law` Legal
       :link: legal
       :link-type: ref

>         Verifies that the project ships with the right legal framework to protect ANSYS, Inc. intellectual property.

Reviewer 3: Again, I question why we don't use "... to protect Ansys's intellectual property."

Author: Thanks for pointing this out, Kathy.

    .. grid-item-card:: :octicon:`gear` Technical
       :link: technical
       :link-type: ref

        Verifies that the project is compliant with ANSYS, Inc. software guidelines and best practices.

.. important::

>     An approval from each part is required to release a project into the public.

Reviewer 3: ```suggestion
    An approval from each of these three parts is required to release a project to the public.
```

When releasing a project to the public, expect to:

* Coordinate with the product line development team, if applicable.
* Maintain the project by means of fixing bugs and providing support for new releases.
* Ensure the good reputation of ANSYS, Inc. in the open source community.

Managerial
^^^^^^^^^^

> The managerial part of the approval process guarantees that direct manager (DM),
> general managers (GM), vice presidents (VP), and chief technolofy officer (CTO)

Reviewer 3: ```suggestion
The managerial part of the approval process guarantees that the direct manager (DM),
general managers (GMs), vice presidents (VPs), and chief technology officer (CTO)
```
Yea! I see you spelled out the titles here. I do not see "CT" here so assume that you can remove it from the earlier reference.

> general managers (GM), vice presidents (VP), and chief technolofy officer (CTO)

Author: ```suggestion
general managers (GM), vice presidents (VP), and chief technology officer (CTO)
```

> are aware about the existence and the status of the project.

Reviewer 3: ```suggestion
are aware of the existence and status of the project.
```

The amount of administrative reviews and approvals depends on the nature of the
project. A project can be classified in to the following categories:

.. grid:: 3
    
    .. grid-item-card:: :octicon:`repo` Documentation projects

>         Documentation projects require direct manager approval, legal

Reviewer 3: Do we use DM instead of direct manager since you defined it earlier? You are using the other abbreviations that you defined earlier. If you don't want to use DM in the text, maybe don't define it earlier?

Author: I finally decided not to use any abbreviations for these names. This makes them more explicit.

        review, and documentation proofing. No source code other than
        documentation files is allowed.

>     .. grid-item-card:: :octicon:`tools` Tools supporting products

Reviewer 3: Would you consider using "Tools projects" so that the classification structure is parallel for all?

Author: Thanks for pointing this out. Let me unify the headers for all of those.

>         Tool projects require direct manager and business unit's GM/VP

Reviewer 3: ```suggestion
        Tool projects require the direct manager and business unit's GM/VP
```

        approval. No product source code is allowed.

>     .. grid-item-card:: :octicon:`container` Libraries supporting products

Reviewer 3: Same comment here. Can the classification be "Library projects"?

        Library projects interfacing and wrapping any ANSYS, Inc. products

>         require approval of the direct manager, product line, product line, and

Reviewer 3: No sure what is meant by "product line"--and the fact that it is repeated!

Author: Good catch Kathy! A product line is the collection of related products that belong to a company.

        the CTO. No product source code is allowed.

Legal
^^^^^

> The legal review and approval guarantees that the whole project is subjected
> and protected by the ANSYS, Inc. legal framework.

Reviewer 3: ```suggestion
The legal review and approval guarantee that the entire project is subjected
and protected by the ANSYS, Inc. legal framework.
```
I would remove "subjected and protected" because I have no idea what is meant by subjected by the Ansys, Inc. legal framework.

Author: Let me rewrite this as:

```
The approval of the legal review guarantees...
```

Also, let me remove the "subjected and..." in favor of:

```
... project is compliant with Ansys legal policies.
```

Does this improve the readability of the text, @PipKat?

Reviewer 3: @jorgepiloto Perfect on both counts. Thanks!

> Start by completing the legal review request form:

Reviewer 3: ```suggestion
Start by completing the legal review request form for open sourcing the code:
```

.. button-link:: https://ansys.sharepoint.com/:w:/r/sites/OpenSourceSoftwareOSSSuperintendence/_layouts/15/Doc.aspx?sourcedoc=%7B3296AD39-79EC-4F42-81C1-1DF988986800%7D&file=Open%20Source%20Policy_Request%20to%20Release%20Code_need%20GM%20sign-off_2021Sep.docx&action=default&mobileredirect=true
    :color: black
    :expand:

    **Open Source Code Release Request Form**

The following checks are required when performing the legal review of the project:

.. card:: |uncheck| The project contains the right licensing

    * The project contains the right license.
    * The contribution does not contain any strong encryption.
    * Ansys official logos and branding images are used in the project.
    * The ANSYS, Inc. copyright appears in the right location as required by the Legal department.

>     * The copyright has the right formatting, that is ``Copyright (C) YYYY ANSYS, Inc.``.

Reviewer 3: ```suggestion
    * The copyright has the right formatting, which is ``Copyright (C) YYYY ANSYS, Inc.``.
```

    * The contribution does not embody any Ansys intellectual property that is not approved for open sourcing.
    * The contribution does not embody any invention for which Ansys has sought or received patent protection.
    * Any third-party open sources included in the contribution have been reviewed for security vulnerabilities and have had their license files included in the repository.

Open source dependencies that are not distributed as part of the project do not
need their licenses included in the Ansys repository. Examples include

> dependent npm modules or Python packages from PyPI.

Reviewer 3: Of course Vale doesn't recognize "npm" but neither do it. Can this be phrased differently?

Author: I agree. This is the Node package manager. Let me rewrite this.

Technical
^^^^^^^^^

> The technical review and approval guarantees that the project follows the best

Reviewer 3: ```suggestion
The technical review and approval guarantees that the project follows the best
```
Again, this is fine if you perceive "technical review and approval" as one entity. Remove the "s" on guarantees if they are to be seen as two different things.

Author: Let me rewrite this as:

```
The approval of the technical review guarantees...
```

and latest software development practices.

The following checks are required when performing the technical review of the project:

> .. card:: |uncheck| The project contains the right metadata information

Reviewer 3: I think this metadata line is a check and requires a period at the end of the sentence.

    
    * The name of the project follows naming convention.
    * The version of the project follows :ref:`Semantic versioning`.
    * The author of the project is ANSYS, Inc.
    * The maintainer of the project is ANSYS, Inc.
    * Contact and support information are provided in the project.
    * :ref:`The \`\`LICENSE\`\` file` is present and compliant with legal requirements.
    * :ref:`The \`\`CONTRIBUTING.md\`\` file` is present.

> .. card:: |uncheck| The project is compliant with PyAnsys style guidelines

Reviewer 3: Same comment about needing a period here.

    * The layout of the project follows the :ref:`Packaging style` guidelines.

>     * :ref:`Testing` guarantees at least 80% of code coverage.

Reviewer 3: ```suggestion
    * :ref:`Testing` guarantees at least 80% code coverage.
```

    * The project follows the :ref:`Documentation style` guidelines.
    * The examples in the source code docstrings are tested.

>     * Documentation examples are presented in the form of a gallery.

Reviewer 3: ```suggestion
    * The documentation examples are presented in the form of a gallery.
```
Suggested change only to keep all the wording for all checks consistent.

    * The package builds properly.

>     * The project contains uses CI/CD with all the :ref:`Required workflows`.

Reviewer 3: ```suggestion
    * The project uses CI/CD with all the :ref:`Required workflows`.
```

    * The CI/CD pipeline generates project :ref:`artifacts`.

.. card:: |uncheck| The GitHub repository is properly secured

>     * Repository is compliant with the :ref:`General configuration`.

Reviewer 3: ```suggestion
    * The repository is compliant with the :ref:`General configuration`.
```

    * :ref:`Branch protection` is enabled.
    * :ref:`Tag protection` is enabled.
    * :ref:`Workflow protection` is enabled.

.. note::

    Contact `support@pyansys.com <mailto:support@pyansys.com>`_ if you require

>     support for being compliant with all previous requirements.

Reviewer 3: ```suggestion
    help meeting all of the previous requirements.
```
